microsoft/testsuites/vm_extensions/vmsnapshot_extension.py

```diff
@@ -100,6 +118,45 @@ def verify_vmsnapshot_extension(
                     pass
                 else:
                     raise e
-            time.sleep(1)
+            time.sleep(60)
             count = count + 1
         assert_that(count, "Restore point creation failed.").is_less_than(10)
+
+    def get_restore_point(
+        self,
+        log: Logger,
+        environment: Environment,
+        resource_group_name: str,
+        restore_point_collection: str,
+        restore_point: str,
+    ) -> int:
+        assert environment.platform
+        platform: AzurePlatform = environment.platform  # type: ignore
+        assert isinstance(platform, AzurePlatform)
+        client = get_compute_client(platform)
+        attempts = 0
+        max_attempts = 15
+        while attempts < max_attempts:
+            response = client.restore_points.get(
+                resource_group_name=resource_group_name,
+                restore_point_collection_name=restore_point_collection,
+                restore_point_name=restore_point,
+                expand=None,
+            )
+            if response.provisioning_state == "Succeeded":
+                log.info(f"restore point {restore_point} created")
+                consistency_mode = response.consistency_mode
+                log.info(f"consistency mode is {consistency_mode}")
+                if (
+                    "FileSystemConsistent" in consistency_mode
+                    or "ApplicationConsistent" in consistency_mode
+                ):
+                    return 0
+                else:
+                    return 1
+            else:
+                log.info(f"rp status is {response.provisioning_state}")
+                attempts += 1
+                if attempts < max_attempts:
```

Comment:
no need this step?

Reply:
just to skip sleep for 1 min at last.
i.e during 14th attempt, if provisioning state is not succeeded then attempts would be incremented to 15, and sleep for a min and then check the exit criteria(while condition) and then simply exits.
So sleeping for a minute is like of no use at the end as we are not checking the status again, so I have put that check.

Reply:
Cannot get the point here. Sleep 60 seconds is too long in any cases. It should check some conditions continually in a loop, once it meets, get out of the loop. The interval in the loop should be less than 5 seconds. So once the condition meets, it waits 5 seconds at most, instead of 60 seconds.

Reply:
reduced the interval to 2 and increased the iteration number.
The above point I was mentioning is : why i have an extra if condition check. That is to skip sleeping for the specified amount of time during the last attempt.